miraclelinux/meta-debian-extended
# avahi: Update version from 0.7-4+deb10u1 to 0.7-4+deb10u2

# Purpose of pull request
This version fixes following CVEs.
- CVE-2023-1981

DLA: DLA 3414-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. Build qemuarm64 image
2. Run qemu image
3. Check avahi-daemon is working
4. Check installed files
5. Check commands of avahi-utils is working
6. Check from host PC that avahi-daemon of qemu environment is working

### Build qemuarm64 image
Add the following to local.conf and build qemuarm64 image.
```
IMAGE_INSTALL_append = " avahi-daemon avahi-utils"
```
```
$ bitbake core-image-minimal
```

### Run qemu image
```
$ runqemu qemuarm64 nographic
```

### Check avahi-daemon is working
Run following commands to check avahi-daemon working.
```
# ps | grep avahi
# /etc/init.d/avahi-daemon status
# /etc/init.d/avahi-daemon restart
```

### Check installed files
```
# find / -name '*avahi*'
```

### Check commands of avahi-utils is working
Run following commands to check each command of avahi-utils works.
```
# avahi-browse -at
# ip address
# avahi-publish -a -R qemu-test.local <eth0-IP-Address> &
# avahi-resolve -4 -n qemuarm64.local
# avahi-resolve -4 -n qemu-test.local
# avahi-resolve -4 -a <eth0-IP-Address>
```

### Check from host PC that avahi-daemon on qemu environment is working
Run following commands on host PC.
```
$ avahi-browse -at | grep qemu
$ ping -c 1 qemuarm64.local
$ ping -c 1 qemu-test.local
```

## Test result
### build
Build succeeded.
```
$ bitbake avahi
Checking Debian Release ...
Sources.xz is not changed.
Force regenerate source code information.
Parsing Debian Sources.xz ...
Parsing recipes: 100% |########################################################################################################################| Time: 0:00:50
Parsing of 1362 .bb files complete (0 cached, 1362 parsed). 2384 targets, 71 skipped, 6 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "ubuntu-20.04"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.7"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "HEAD:706cb4b3a04ef1c79838a92fe2d0eb428e08b1e7"
meta-debian-extended = "update-avahi_0.7-4+deb10u2:b482ae7926d6c38f04f708857030653166eea330"
meta-emlinux         = "HEAD:42da1eaabe2b445010cf4f0181eb6edaf811a3ee"
meta-emlinux-private = "HEAD:b290dafea47dfbd8df7ee9b04af38c9b73c87b68"

Initialising tasks: 100% |#####################################################################################################################| Time: 0:00:04
Sstate summary: Wanted 469 Found 0 Missed 469 Current 0 (0% match, 0% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2151 tasks of which 0 didn't need to be rerun and all succeeded.
```

### Package test
The avahi-daemon is running when system is started.
```
EMLinux 2.7 qemuarm64 /dev/ttyAMA0

qemuarm64 login: root
root@qemuarm64:~# ps | grep avahi-daemon
 1631 avahi     4824 S    avahi-daemon: running [qemuarm64.local]
 1632 avahi     4696 S    avahi-daemon: chroot helper
 1933 root      3220 S    grep avahi-daemon
```

The files installed by avahi-daemon and avahi-utils packages appear to be installed to correct path.
```
root@qemuarm64:~# find / -name '*avahi*'
/lib/systemd/system/avahi-daemon.socket
/lib/systemd/system/avahi-daemon.service
find: /proc/4347: No such file or directory
find: /proc/4377: No such file or directory
find: /proc/4378: No such file or directory
find: /proc/4379: No such file or directory
find: /proc/4380: No such file or directory
/usr/lib/libavahi-client.so.3.2.9
/usr/lib/libavahi-client.so.3
/usr/lib/libavahi-common.so.3.5.3
/usr/lib/libavahi-core.so.7.0.2
/usr/lib/libavahi-common.so.3
/usr/lib/libavahi-core.so.7
/usr/sbin/avahi-daemon
/usr/bin/avahi-resolve-address
/usr/bin/avahi-browse
/usr/bin/avahi-publish
/usr/bin/avahi-publish-address
/usr/bin/avahi-set-host-name
/usr/bin/avahi-resolve
/usr/bin/avahi-resolve-host-name
/usr/bin/avahi-publish-service
/usr/bin/avahi-browse-domains
/usr/share/avahi
/usr/share/avahi/avahi-service.dtd
/run/avahi-daemon
/etc/rc2.d/S21avahi-daemon
/etc/rc1.d/K19avahi-daemon
/etc/systemd/system/multi-user.target.wants/avahi-daemon.service
/etc/systemd/system/sockets.target.wants/avahi-daemon.socket
/etc/rc5.d/S21avahi-daemon
/etc/rc4.d/S21avahi-daemon
/etc/rc3.d/S21avahi-daemon
/etc/rc6.d/K19avahi-daemon
/etc/init.d/avahi-daemon
/etc/dbus-1/system.d/avahi-dbus.conf
/etc/avahi
/etc/avahi/avahi-daemon.conf
/etc/rc0.d/K19avahi-daemon
```

Each command of avahi-utils is working.
```
root@qemuarm64:~# avahi-browse -at
+   eth0 IPv4 qemuarm64                                     _ssh._tcp            local
+   eth0 IPv4 qemuarm64                                     _sftp-ssh._tcp       local
root@qemuarm64:~# ip address
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast qlen 1000
    link/ether 52:54:00:12:34:02 brd ff:ff:ff:ff:ff:ff
    inet 192.168.7.2/24 brd 192.168.7.255 scope global eth0
       valid_lft forever preferred_lft forever
root@qemuarm64:~# avahi-publish -a -R qemu-test.local 192.168.7.2 &
root@qemuarm64:~# Established under name 'qemu-test.local'
root@qemuarm64:~# avahi-resolve -4 -n qemuarm64.local
qemuarm64.local	192.168.7.2
root@qemuarm64:~# avahi-resolve -4 -n qemu-test.local
qemu-test.local	192.168.7.2
root@qemuarm64:~# avahi-resolve -4 -a 192.168.7.2
192.168.7.2	qemuarm64.local
```

The host-name configured in qemu environment can resolved names correctly on host PC.
```
~$ PS1='\w$ '
~$ avahi-browse -at | grep qemu
+   tap0 IPv4 qemuarm64                                     SSH Remote Terminal  local
+   tap0 IPv4 qemuarm64                                     SFTP File Transfer   local
~$ ping -c 1 qemuarm64.local
PING qemuarm64.local (192.168.7.2) 56(84) bytes of data.
64 bytes from 192.168.7.2 (192.168.7.2): icmp_seq=1 ttl=64 time=5.22 ms

--- qemuarm64.local ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 5.218/5.218/5.218/0.000 ms
~$ ping -c 1 qemu-test.local
PING qemu-test.local (192.168.7.2) 56(84) bytes of data.
64 bytes from 192.168.7.2 (192.168.7.2): icmp_seq=1 ttl=64 time=0.343 ms

--- qemu-test.local ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.343/0.343/0.343/0.000 ms
```
